STYLE-GUIDE.md

```diff
@@ -0,0 +1,134 @@
+> environments are welcomed and valued members of the community and their insights help shape the software.
+
+OpenLMIS is built and used by those in low-resource settings:
+
+* Internet is often slow and intermittent.  Features should be designed with these limitations in mind.  For example, most 
+work-flows should be optimized for slow internet and even work-flows with periods of non-connectivity.  
+Administrative screens however can often take shortcuts and assume that their users will have better internet 
+connectivity.
+* Processes not only vary and need to be configurable by program and implementation, they oftentimes are  
+used in parallel or supplement traditional paper processes.  Data collection and forms should strive to be able to 
+be configurable to match a paper form and be able to restore it historically.
+* Screens are often older and come with lower resolutions than the latest and greatest.  800x600 px screens are not uncommon.  Additionally, many work-flows that would be used by someone at the last mile will be used by someone with a
+smaller tablet or even a phone.
+* Scalability for OpenLMIS is the capability of use in large hospitals to community health 
+workers nation wide.  The workflow from data collection, processing through to report delivery should be designed and implemented for thousands of users with thousands of physical facilities.
+* Security is important for OpenLMIS to be trusted to run nation-wide government supply chains to NGO initiatives.  
+A role-based security system contains users to see and do only what is required for their role.  Care should be given
+in designing features and running implementations to keep OpenLMIS secure.
+
+### <a name="configurable"></a>Configurable
+> OpenLMIS flexibly supports the varied needs of low-resource health supply chains. OpenLMIS strives to be 
+> designed so that countries can configure and use the software with minimal training and technical capacity.
+
+Supply chains vary.  Reporting requirements, process differences, language, and even the look and feel need to be as 
+configurable as is reasonable for OpenLMIS to continue to deliver on its mission.  In order to accomplish this, OpenLMIS
+contributions need to at a minimum continue to deliver:
+
+* Language - Language tags allow messages/UI/email/API/etc to be translated into many different languages and allows 
+the user to switch the language displayed easily.  OpenLMIS has standardized development in English for consistency
+and supports translation projects as the opportunity arises.
+* Dates - Date formatting also varies by locality.  As such any date or time printed should allow for custom 
+formatting.
+* **Programs** allow for OpenLMIS to configure the vertical supply chains present in many low- and middle-income 
+countries independently.  e.g. a Malaria program may collect different data by different people than an HIV/AIDS 
+program.
```

Comment:
Both sound good. I have no more concern to the style guide.


On 13 February 2016 at 05:04, Josh Zamor <notifications@github.com> wrote:

> In STYLE-GUIDE.md
> <https://github.com/OpenLMIS/open-lmis/pull/90#discussion_r52798100>:
>
> > +### <a name="configurable"></a>Configurable
> > +> OpenLMIS flexibly supports the varied needs of low-resource health supply chains. OpenLMIS strives to be
> > +> designed so that countries can configure and use the software with minimal training and technical capacity.
> > +
> > +Supply chains vary.  Reporting requirements, process differences, language, and even the look and feel need to be as
> > +configurable as is reasonable for OpenLMIS to continue to deliver on its mission.  In order to accomplish this, OpenLMIS
> > +contributions need to at a minimum continue to deliver:
> > +
> > +* Language - Language tags allow messages/UI/email/API/etc to be translated into many different languages and allows
> > +the user to switch the language displayed easily.  OpenLMIS has standardized development in English for consistency
> > +and supports translation projects as the opportunity arises.
> > +* Dates - Date formatting also varies by locality.  As such any date or time printed should allow for custom
> > +formatting.
> > +* **Programs** allow for OpenLMIS to configure the vertical supply chains present in many low- and middle-income
> > +countries independently.  e.g. a Malaria program may collect different data by different people than an HIV/AIDS
> > +program.
>
> Vaccines in my view are just another program. They share mostly the same
> nature as other verticals within a country. They usually come with more of
> an automated resupply process, quicker resupply loops and of course - the
> cold chain.
>
> Equipment to me breaks down so far into 2 categories: cold chain equipment
> that's mostly either actively or passively informing the vaccine program,
> and more of the asset management side that's concerned with lab type
> equipment. I'm more familiar with the vaccine side of things, and pretty
> confident that the lab-equipment side of things aren't being used (plus
> there is an open source project for managing lab equipment). I know there
> is some consolidation that's needed with cold-chain equipment in how it can
> inform vaccine type programs between the push and pull systems.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenLMIS/open-lmis/pull/90/files#r52798100>.
>



-- 
Jeff Xiong
http://gigix.thoughtworkers.org